mozilla-services/socorro
# Fix bug 1399305: Script deletion and creation of S3 bucket and ES indices in development environment.

This does 3 things:

1. Adds `docker/run_recreate_s3_buckets.sh`, which runs during `make dockersetup` and deletes and re-creates the `dev_bucket` S3 bucket used during development.
2. Adds a new Socorro app called `clear_es_indices`, run during `make dockersetup`, that clears all existing Elasticsearch indexes.
3. Adds a new Socorro app called `create_recent_es_indices`, run during `make dockerupdatedata`, that creates Elasticsearch indices for the next two weeks. This fixes the issues with the homepage and other views that pop up during local development when they're missing.

We don't have great support for testing changes to Elasticsearch, so I filed [bug 1429267](https://bugzilla.mozilla.org/show_bug.cgi?id=1429267) as a followup to add better scaffolding such that we could add automated tests for scripts like these.

To test this:
1. Start with a Socorro installation that has processed some crashes.
2. Run `make dockersetup`. You should see logging output that looks something like this:

   ```
   Dropping and recreating S3 bucket...
   S3 container bucket is dev_bucket
   delete: s3://dev_bucket/v1/dump/01557534-e7c7-4936-970d-86f550180107
   (...snip...)
   ```

   You should also see a log line like this, indicating that the existing ES indices have been deleted:

   ```
   2018-01-10 01:22:23,085 DEBUG - clear_es_indices -  - MainThread - Deleted elasticsearch indices.
   ```
3. Run `make dockerupdatedata`. You should see logging output indicating that new ES indices have been created:

   ```
   2018-01-11 21:13:01,754 INFO - create_recent_es_indices -  - MainThread - Created new elasticsearch index: socorro201801
   2018-01-11 21:13:01,812 INFO - create_recent_es_indices -  - MainThread - Created new elasticsearch index: socorro201802
   2018-01-11 21:13:01,893 INFO - create_recent_es_indices -  - MainThread - Created new elasticsearch index: socorro201803
   ```
4. Run `docker-compose up webapp` and visit http://localhost:8000/home/product/Firefox. You should see a graph on the front page, although it may contain no data (you should not see an error message, which indicates the indices were not found).
